tapdb/multiverse.go

```diff
@@ -944,6 +1021,14 @@ func (b *MultiverseStore) UpsertProofLeaf(ctx context.Context,
 	b.proofCache.delProofsForAsset(id)
 	b.leafKeysCache.wipeCache(idStr)
 
+	// Notify subscribers about the new proof leaf, now that we're sure we
+	// have written it to the database. But we only care about transfer
+	// proofs, as the events are received by the custodian to finalize
+	// inbound transfers.
+	if id.ProofType == universe.ProofTypeTransfer {
+		b.eventDistributor.NotifySubscribers(leaf.RawProof)
+	}
```

Comment:
While I agree in general in terms of separation of concern, this would then notify for each new issuance proof we sync from a universe, which could be a lot. So performance wise I think it's a reasonable tradeoff for now. I also made things more clear by renaming `eventDistributor` to `transferProofDistributor` since we only use it for that purpose right now.

Reply:
I like the rename! Sets expectations up nicely.